applications/apophenia.md

```diff
@@ -0,0 +1,267 @@
+5. Method: a supported way with which to get the data (ex: TEE Contracts or Offchain Worker)
+
+> Note: The use of a topic is to allow for multiple feeds to be registered for the same context. For example, there could be multiple feeds for the `dot_usd_price` topic, each with a different source and/or method.
+
+Registering a feed generates a unique key that can be used to consume the feed.
+
+Registering a feed may also require a deposit (stake), which can be used to slash the feed if it is found to be malicious.
+
+##### Feed Consumption
+
+A feed can be consumed via RPC calls currently, however, the addition of an XCM interface is of high priority in the roadmap. 
+
+The consumption process is simple, the client (i.e. actor account consuming the feed) provides the following:
+
+1. Feed Topic: an identifier of the feed's context (ex: `dot_usd_price`)
+2. Feed Key (optional): the ID of the feed to consume from (ex: `...`)
+3. Aggregation (optional): the aggregation method to use (ex: `median`). This can be used to aggregate multiple feeds into a single value.
+
+> Note: Topics are also used to determine if a certain feed is returing values which are out of the expected range. For example, if a feed is returning a value that is too high or too low compared to the median, it can be flagged as invalid. This can be used to prevent malicious actors from manipulating the data and apply slashing.
+
+#### Architecture
+
+The main components of the Apophenia architecture are:
+
+- **Oracle Pallet**
+- **Feed Registry Pallet**
+- **Oracle Client**
+- **Oracle Offchain Worker**
+- **TEE and Smart Contract Pools** (stretch)
+
+![Apophenia Architecture](https://github.com/subslice/oracle-parachain/blob/main/images/Paraxiom%20Pallets%20Overview.png?raw=true)
+
+### Ecosystem Fit
+
+There exists a need for a native Oracle solution within the DotSama ecosystem. Apophenia aims to fill this gap.
+
+The target audience includes parachains, smart contracts and individual accounts can register new feeds and request data from them.
+
+While there are pallets and integrations within some Parachains currently (ex: Dia on Astar or Chainlink's pallet), it remains an added complexity for the end user. Apophenia aims to provide a simple and easy to use solution for all, especially chains within the system.
+
+Apophenia also aims to provide data aggregation functionality to increase both data quality and security. This is a feature that can make Apophenia a unique solution.
+
+In regards to alternatives, we have observed that most of the [projects under Oracle are inactive](https://wiki.polkadot.network/docs/build-open-source). For those that are active, here's a quick comparison:
+
+* interBTC: specialized for Bitcoin quotes, we intend to be a generalized oracle solution instead.
+* DIA: current solution available on Astar as ink! smart contracts. Like Laminar and Chainlink. data is pushed on-chain by centralized (and usually manually verified) offchain services.
```

Comment:
> Like Laminar and Chainlink. data is pushed on-chain by centralized offchain services

I'm not sure if that's true, at least for ChainLink. From their [docs](https://chain.link/education/blockchain-oracles#decentralized-oracles):

> A Decentralized Oracle Network, or DON for short, combines multiple independent oracle node operators and multiple reliable data sources to establish end-to-end decentralization.
> Even more, many Chainlink DONs, such as [Chainlink Price Feeds](https://data.chain.link/), incorporate [three layers of decentralization](https://blog.chain.link/levels-of-data-aggregation-in-chainlink-price-feeds/)—at the data source, individual node operator, and oracle network levels—to eliminate any single point of failure.
> ![image](https://github.com/w3f/Grants-Program/assets/5393704/ff09aa79-c8b7-4764-a18e-115b9c44c33d)

Your approach seems to be very similar to the ChainLink approach. Could you clarify on that?

Reply:
(the section was removed - resolving this)